### PR TITLE
Add remote_precedence parameter to dev_package_deps

### PR DESCRIPTION
**Pull request by maksymiuks**
As discussed in the https://github.com/r-lib/remotes/issues/732 it's a PR with suggested implementation.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Thanks! Hmmm, does this really help people install a package with the CRAN dependencies? How would one do that?

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 * `install_github()` now uses credentials from the git credential store,
   if `GITHUB_PAT` and `GITHUB_TOKEN` are not set.
 * Remotes field accepts explicit package names: `<pkgname>=<type>::<username>/<repo>` (#719, @heavywatal).
+* Add `remote_precedence` parameter to `dev_package_deps()` and `combine_remote_deps()` that allows
```
Comment (gaborcsardi) [inline NEWS.md:7]: `combine_remote_deps()` is not exported, we should not mention it here.

File: NEWS.md [hunk 2]
```diff
@@ -4,6 +4,9 @@
...
+  user to choose whether remote sources should have a priority over cran sources. Parameter defaults
+  to TRUE for backward compatibility reasons. (#732, @maksymiuks)
```
Comment (gaborcsardi) [inline NEWS.md:9]: It is not "just" for backward compatibility, but rather because that's the usual case. So I would just remove this last sentence.

**Review by gaborcsardi — CHANGES_REQUESTED** [review]

**Comment by gaborcsardi** [comment]
> it will be helpful in any case when for some reason we need to update our dependencies to the version available on CRAN, disregarding remotes if possible.

Well, except that if the non-CRAN version is already installed, then it will not "update" that to the CRAN version.

**Comment by maksymiuks** [comment]
I agree word "update" wasn't really a fortunate one. You are right, if the developer stores everything in one library then yes it won't be updated, as probably remote is ahead of CRAN so it will be skipped by the design of this function. But if one has isolated libraries for testing and properly sets the search path, or if it's a ci/cd scenario, it will do its job as the `update` is a very neat feature for installing the package's dependencies in a controlled way.

File: NEWS.md (hunk 2 above)
Comment (maksymiuks) [reply NEWS.md:9]: right, done!

**Comment by gaborcsardi** [comment]
Thanks!

**Pull request merged**